ivy/functional/frontends/numpy/ndarray/ndarray.py

```diff
@@ -336,6 +336,9 @@ def __float__(
     ):
         return ivy.array(ivy.reshape(self._ivy_array, -1), dtype=ivy.float64)[0]
 
+    def __str__(self):
+        return self._ivy_array.__str__()
```

Comment:
I'm not sure this is correct, as `self` is a NumPy Array it would not contain an attribute called `_ivy_array.`

Reply:
Hey! Thanks for the review

`self._ive_array` is the underlying array that is being used, the whole `self` object includes every detail, but for the `__str__` function we only need the array to be converted to string

Also, when I tried to call `str(self)` it got stuck inside an infinite recursion.

This is my opinion, let me know what the final version should be here and I will be happy to update accordingly

Reply:
As this is the Ivy Frontend, what we expect to be input are NDarrays from Numpy. Therefore it would not contain an `_ivy_array` attribute. So this current implementation wouldn't work. As this simply returns a string I think you would only need to use Numpy to return the string object.

Reply:
If you have a look at other functions, e.g. `__int__` or `__float__` they are also using `_ivy_array` attribute inside them
I think this is why I am confused

And calling `str(self)` won't work as I already mentioned and got stuck in an infinite recursion 😅

Reply:
@jkeane508

Reply:
Hi, I've actually looked into this (I wasn't super up to speed with how this module works).

But basically, the issue is this:

The ivy __str__() method return a different string than Numpy

`ivy.array([1]).__str__()`
`'ivy.array([1])'`

`np.array([1]).__str__()`
`'[1]'`

So in order to align behavior with NumPy we would need to format the string returned using the Ivy Method:

`ivy.array([1]).__str__()[10:-1]`

Hope this makes sense :)

Reply:
@jkeane508 I tried to make the function more readable and also called it in `__repr__` to avoid using the same line of codes and, instead use functions

Let me know what you think!